pandas/core/frame.py

```diff
@@ -1336,7 +1343,12 @@ def to_numpy(self, dtype=None, copy: bool = False) -> np.ndarray:
         array([[1, 3.0, Timestamp('2000-01-01 00:00:00')],
                [2, 4.5, Timestamp('2000-01-02 00:00:00')]], dtype=object)
         """
-        result = np.array(self.values, dtype=dtype, copy=copy)
+        arr = [
+            self[c].to_numpy(dtype=dtype, na_value=na_value, **kwargs)
+            for c in self.columns
+        ]
```

Comment:
you can use `self._iter_column_arrays()` here, to avoid creating intermediate Series objects

Reply:
Although, that also returns numpy arrays, which makes it more complicated to support na_value for those ..